mne/viz/epochs.py

```diff
@@ -65,74 +78,292 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+                    ylims[ch_type][1] = this_vmax
+
+    # plot
+    figs = list()
+    for (epochs_, ch_type, ax_name, name, data, overlay_times, vmin, vmax,
+         ts_args) in groups:
+        vmin, vmax = vmins.get(ch_type, vmin), vmaxs.get(ch_type, vmax)
+        these_axes = axes[ax_name] if isinstance(axes, dict) else axes
+        axes_dict = _prepare_epochs_image_axes(these_axes, fig, colorbar,
+                                               evoked)
+        if "ylim" not in ts_args:
+            ts_args["ylim"] = ylims[ch_type]
+        title = ax_name if isinstance(axes, dict) else name
+        this_fig = _plot_epochs_image(
+            epochs_, data, vmin=vmin, vmax=vmax, colorbar=colorbar, show=False,
+            unit=units[ch_type], ch_type=ch_type, cmap=cmap,
+            axes_dict=axes_dict, title=title, overlay_times=overlay_times,
+            evoked=evoked, ts_args=ts_args)
+        figs.append(this_fig)
+
+    plt_show(show)
+    return figs
+
+
+def _get_picks_and_types(picks, ch_types, group_by, combine):
+    """Pack picks and types into a list. Helper for plot_epochs_image."""
+    if group_by is None:
+        if combine is not None:
+            picks = [picks]
+        return picks, ch_types, ch_types
+    elif group_by == "type":
+        all_picks, all_ch_types = list(), list()
+        for this_type in set(ch_types):
+            these_picks = picks[np.array(ch_types) == this_type]
+            all_picks.append(these_picks)
+            all_ch_types.append(this_type)
+        names = all_ch_types  # only differs for dict group_by
+    elif isinstance(group_by, dict):
+        names = list(group_by.keys())
+        all_picks = [group_by[name] for name in names]
+        for name, picks_ in group_by.items():
+            n_picks = len(picks_)
+            if n_picks < 2:
+                raise ValueError(" ".join(
+                    (name, "has only ", str(n_picks), "sensors.")))
+        all_ch_types = list()
+        for picks_, name in zip(all_picks, names):
+            this_ch_type = list(set((ch_types[pick] for pick in picks_)))
+            n_types = len(this_ch_type)
+            if n_types > 1:  # we can only scale properly with 1 type
+                raise ValueError(
+                    "Roi {} contains {} sensor types!".format(
+                        name, n_types))
+            all_ch_types.append(this_ch_type[0])
+            names.append(name)
+    else:
+        raise ValueError("If `group_by` is not None, it must be a dict "
+                         "or 'type', got " + type(group_by))
+    return all_picks, all_ch_types, names  # all_picks is a list of lists
+
+
+def _pick_and_combine(epochs, combine, all_picks, all_ch_types, scalings,
+                      names):
+    """Pick and combine epochs image. Helper for plot_epochs_image."""
+    to_plot_list = list()
+    tmin = epochs.times[0]
+
+    if combine is None:
+        epochs = epochs.copy().load_data()
```

Comment:
And this is potentially very slow on big datasets. You're only picking one channel right? Maybe it'll be worth the effort to load the data once and use that to construct the epochs individually from scratch.